spinalcordtoolbox/moco.py

```diff
@@ -865,3 +868,209 @@ def spline(folder_mat, nt, nz, verbose, index_b0 = [], graph=0):
+
+    # Copying input data to tmp folder
+    sct.printv('\nCopying input data to tmp folder and convert to nii...', param.verbose)
+    convert(param.fname_data, os.path.join(path_tmp, file_data))
+    if param.fname_mask != '':
+        convert(param.fname_mask, os.path.join(path_tmp, file_mask), verbose=param.verbose)
+
+    # Build absolute output path and go to tmp folder
+    curdir = os.getcwd()
+    path_out_abs = os.path.abspath(param.path_out)
+    os.chdir(path_tmp)
+
+    # Load data and get dimensions
+    sct.printv('\nGet dimensions of data...', param.verbose)
+    im_data = Image(file_data)
+    nx, ny, nz, nt, px, py, pz, pt = im_data.dim
+    sct.printv('  ' + str(nx) + ' x ' + str(ny) + ' x ' + str(nz) + ' x ' + str(nt), param.verbose)
+
+    sct.printv('\nData were acquired in interleaved mode.', param.verbose)
+    sct.printv('Splitting data into two datasets along SI direction'
+        ' (even and odd slices) and run moco separately...', param.verbose)
+    orig_orientation = split_to_even_and_odd(im_data, file_data)
+
+    # Split mask if was passed
+    if param.fname_mask != '':
+        mask_data = Image(file_mask)
+        split_to_even_and_odd(mask_data, file_mask)
+
+    # ==================================================================================================================
+    # Run moco only on even slices
+    # ==================================================================================================================
+    sct.printv('\nStarting moco on even slices...', param.verbose)
+    # Update fields in param (because used later in another function, and param class will be passed)
+    param.fname_data = 'data_even.nii'
+    param.path_out = ''
+    if param.fname_mask != '':
+        param.fname_mask = 'mask_even.nii'
+    im_data_even = moco_wrapper(param)
+    if param.output_motion_param:
+        # Rename .nii and tsv moco parameters after moco (add _even suffix)
+        for name in file_moco_params_x, file_moco_params_y, file_moco_params_csv:
+            sct.mv(name, sct.add_suffix(name, '_even'))
+
+    # ==================================================================================================================
+    # Run moco only on odd slices
+    # ==================================================================================================================
+    sct.printv('\nStarting moco on odd slices...', param.verbose)
+    # Update field in param (because used later in another function, and param class will be passed)
+    param.fname_data = 'data_odd.nii'
+    param.path_out = ''
+    if param.fname_mask != '':
+        param.fname_mask = 'mask_odd.nii'
+    im_data_odd = moco_wrapper(param)
+    if param.output_motion_param:
+        # Rename .nii and tsv moco parameters after moco (add _odd suffix)
+        for name in file_moco_params_x, file_moco_params_y, file_moco_params_csv:
+            sct.mv(name, sct.add_suffix(name, '_odd'))
```

Comment:
this duplicated code could easily be called by a private function (and not be duplicated)